docs/customization/api.rst

```diff
@@ -95,3 +95,198 @@ The next step is to modify the security configuration in ``config/packages/secur
+
+            $data = $this->objectNormalizer->normalize($object, $format, $context);
+
+            return $data;
+        }
+
+        public function supportsNormalization($data, $format = null, $context = []): bool
+        {
+            return $data instanceof ProductInterface;
+        }
+    }
+
+And now let's declare it's service in config files.
+
+.. code-block:: xml
+
+    <service id="App\Serializer\ProductSerializer">
+            <argument type="service" id="api_platform.serializer.normalizer.item" />
+            <tag name="serializer.normalizer" />
+    </service>
+
+Then we can add the new field.
+
+.. code-block:: php
+
+    //...
+    $data = $this->objectNormalizer->normalize($object, $format, $context);
+
+    $data['additionalText'] = 'your custom text or logic that will be added to this field.';
+
+    return $data;
+    //...
+
+Now your response should be extended with new field
+
+.. code-block:: javascript
+
+    {
+        //...
+        "id": 123,
+        "code": "product_code",
+        "variants": [
+            "/api/v2/shop/product-variants/product-variant-0",
+        ],
+        "additionalText": "my additional field with text",
+        //...
+    }
+
+Removing a field from response
+==============================
+
+Let's say that for some reason you want to remove some field from serialization.
+Your possible solution could be that you use serialization groups.
+Those will limit the fields from your resource, according to serialization groups that you will choose.
+
+.. tip::
+
+    Read more about API Platform `serialization groups <https://api-platform.com/docs/core/serialization/#using-serialization-groups>`_
+
+But if you want to remove the field by utilising serializer, first step is to create a class as in ``Adding a field from response`` and register it's service.
+
+Let's assume that ``Product`` resource returns
+
+.. code-block:: javascript
+
+    {
+        //...
+        "id": 123,
+        "code": "product_code",
+        "variants": [
+            "/api/v2/shop/product-variants/product-variant-0",
+        ],
+        "translations": {
+            "en_US": {
+              "@id": "/api/v2/shop/product-translations/123",
+              "@type": "ProductTranslation",
+              "id": 123,
+              "name": "product name",
+              "slug": "product-name"
+        }
+    }
+
+Then let's say you want to remove ``translations``. We can do it by adding
+
+.. code-block:: php
+
+    //...
+    $data = $this->objectNormalizer->normalize($object, $format, $context);
+
+    unset($data['translations']); // removes `translations` from response
+
+    return $data;
+    //...
+
+Now your response fields should look like this
+
+.. code-block:: javascript
+
+    {
+        //...
+        "id": 123,
+        "code": "product_code",
+        "variants": [
+            "/api/v2/shop/product-variants/product-variant-0",
+        ]
+    }
+
+Renaming a field from response
```

Comment:
It should be achieved by setting `serialized-name` attribute in serialization config, similar as: https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/ApiBundle/Resources/config/serialization/Commands/AddProductReview.xml#L31